lib/pbench/server/database/models/datasets.py

```diff
@@ -1151,9 +1128,50 @@ def setvalue(dataset: Dataset, key: str, value: Any, user: Optional[User] = None
                     dataset, key, value, f"date/time before {maximum:%Y-%m-%d}"
                 )
             target += __class__.ONE_DAY
-            v = f"{target:%Y-%m-%d}"
+            return f"{target:%Y-%m-%d}"
         else:
-            v = value
+            return value
+
+    @staticmethod
+    def setvalue(dataset: Dataset, key: str, value: Any, user: Optional[User] = None):
+        """
+        Create or modify an existing metadata value. This method supports
+        hierarchical dotted paths like "dashboard.seen" and should be used in
+        most contexts where client-visible metadata paths are used.
+
+        This will create a nested JSON structure (represented as a Python dict)
+        as necessary as it descends the hierarchy. For example, if you assign
+        "a.b.c" a value of "bar", and then query "a", you'll get back
+        "a": {"b": {"c": "bar"}}}
+
+        We handle two keys specially:
+
+            dataset.name: The primary resource name of a dataset, which can be
+                modified by the dataset's owner.
+            server.deletion: The expected automatic deletion date of the
+                dataset. It can be modified by the owner so long as the new
+                value is a valid date that falls within the maximum server
+                retention period from the dataset's upload timestamp.
+
+        Args:
+            dataset: Associated Dataset
+            key: Lookup key (including hierarchical dotted paths)
+            value: Value to be assigned to the specified key
+        """
+        if not Metadata.is_key_path(key, Metadata.METADATA_KEYS):
+            raise MetadataBadKey(key)
+        keys = key.lower().split(".")
+        native_key = keys.pop(0)
+        found = True
+
+        # Setting the dataset name is a direct modification to the Dataset SQL
+        # column, so do that first and exit without touching the Metadata
+        # table.
+        v = __class__.validate(dataset, key, value)
+        if key == __class__.DATASET_NAME:
```

Comment:
It looks to me like line 1170 should precede the block comment, since the comment applies to the following line and 1170 applies to all cases.

Reply:
I knew you were going to say that. But I didn't know it until I'd already committed the code and I didn't feel like fixing it then.